### PR TITLE
Add tests for `CreateBucket`

### LINKED ISSUE #361: Test `CreateBucket` Method in BucketRuntime Service
# Summary
Test `CreateBucket` Method in BucketRuntime Service

## Description
Test the `CreateBucket` method in the `BucketRuntime` service to ensure that it creates a new bucket as expected. Cover various scenarios including valid inputs, invalid inputs, and boundary cases. Verify that the method handles errors properly and provides appropriate responses.

### Instructions:
1. Install the necessary CRDs in the test environment.
2. Write test cases to validate the `CreateBucket` method functionality.
3. Ensure that the tests only patch the `rook` resources for testing purposes.
4. Test cases should include valid bucket creation, creation with invalid parameters, and concurrent creation attempts.
5. Verify that error responses are accurate and informative.

### PR DESCRIPTION
**Pull request by Rohit-0505**
- Add suite test for Bucket
- Add tests for `CreateBucket` function Fixes #361

### CODE REVIEW

File: ori/bucket/server/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+		},
+		ErrorIfCRDPathMissing: true,
+	}
+
+	testEnvExt = &envtestutils.EnvironmentExtensions{
+		APIServiceDirectoryPaths: []string{
+			modutils.Dir("github.com/onmetal/onmetal-api", "config", "apiserver", "apiservice", "bases"),
+		},
+		ErrorIfAPIServicePathIsMissing: true,
+	}
```
Comment (afritzler) [inline ori/bucket/server/suite_test.go:109]: We don't need the `onmetal-api` here and hence can remove this here.

File: ori/bucket/server/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		Port:         testEnvExt.APIServiceInstallOptions.LocalServingPort,
+		CertDir:      testEnvExt.APIServiceInstallOptions.LocalServingCertDir,
+	})
+	Expect(err).NotTo(HaveOccurred())
+
+	By("starting the onmetal-api aggregated api server")
+	Expect(apiSrv.Start()).To(Succeed())
+	DeferCleanup(apiSrv.Stop)
+
+	Expect(envtestutils.WaitUntilAPIServicesReadyWithTimeout(apiServiceTimeout, testEnvExt, k8sClient, scheme.Scheme)).To(Succeed())
```
Comment (afritzler) [inline ori/bucket/server/suite_test.go:140]: As stated above: as we are only working against the Rook API here we don't need the `onmetal-api` server in this test suite.

**Review by afritzler — CHANGES_REQUESTED** [review]

File: ori/bucket/server/suite_test.go (hunk 2 above)
Comment (sujeet01) [reply ori/bucket/server/suite_test.go:140]: Since  [here](https://github.com/onmetal/cephlet/blob/6981634ddcdc0f4bee1f36dcbe2da03f8c4ac95a/ori/bucket/server/suite_test.go#L212) we need to create a `BucketClass`  which is of type `github.com/onmetal/onmetal-api/api/storage/v1alpha1`, hence we used onmetal api server.

**Review by afritzler — APPROVED** [review]
